### PR TITLE
Developer doc clean-up 

### PR DESCRIPTION
**Pull request by katriendg**
The developer guide still had some notes about CLI version bug, which is now fixed. Updated and clean-up a bit further to simplify doc.

### CODE REVIEW

File: deployment/deploy-dev.md [hunk 1]
```diff
@@ -47,27 +39,19 @@ Go to the `deployment` folder and run `pwsh ./deploy-az-dev-bootstrapper.ps1 -Ap
 
 Subsequent deployments can be run as follows.
```
Comment (suneetnangia) [inline deployment/deploy-dev.md:40]: I loved this change when you did that initially, so much easier to iterate over changes in a dev loop.

**Review by suneetnangia — COMMENTED** [review]
Thanks, it looks good, added some minor comments for your review.

File: deployment/deploy-dev.md [hunk 2]
```diff
@@ -2,38 +2,30 @@
...
-Go to the `deployment` folder and run `pwsh ./deploy-az-dev-bootstrapper.ps1 -ApplicationName <short-name>`
+In a PowerShell environment, go to `deployment` folder and run `./deploy-az-dev-bootstrapper.ps1 -ApplicationName <short-name>`
 
 ## The main functions in the script
 
 1. Deploy infrastructure with Bicep, the script deploys AKS cluster.
     * AKS
     * VNET
-    * Deploy Squid proxy in each layer
+    * Squid proxy in cluster
```
[the remark this answers is not in the document]
Comment (katriendg) [reply deployment/deploy-dev.md:24]: That's a good point, the default devloop should not need to deploy squid and Arc. I'll remove this for now, it's going to change potentially in future.

File: deployment/deploy-dev.md [hunk 3]
```diff
@@ -2,38 +2,30 @@
...
-This environment is used to setup a fast developer environment which means that Arc and Flux is not included in the environment and only a single level (single cluster) is deployed.
+This flow is used to setup a developer environment in Azure without Azure Arc and Flux. Additionally, only one single level (single AKS cluster) is deployed. Everything is executed from the local developer machine.
 
 ## Prerequisites on developer machine
 
 - PowerShell on Windows, or PowerShell Core on Linux/MacOS
 - .NET 6.0
 - .NET Core 3.1
-- Azure CLI 2.37+
+- Azure CLI 2.40+
```
[the remark this answers is not in the document]
Comment (katriendg) [reply deployment/deploy-dev.md:12]: I've added it in the meantime in my current branch (data plane), but will add it here to complete in one go.

File: deployment/deploy-dev.md [hunk 4]
```diff
@@ -2,38 +2,30 @@
...
 - Docker
 
 ## How to execute it
 
-### Windows
-
-Go to the `deployment` folder and run `.\deploy-az-dev-bootstrapper.ps1 -ApplicationName <short-name>`
```
[the remark this answers is not in the document]
Comment (katriendg) [reply deployment/deploy-dev.md:19]: Agree, will add another task to the board and leave that for another iteration.

**Pull request merged**